test/integration_test.go

```diff
@@ -0,0 +1,120 @@
+package test
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/defaults"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/gruntwork-io/terragrunt/cli"
+)
+
+// hard-code this to match the test fixture for now
+const (
+	TERRAFORM_REMOTE_STATE_S3_REGION      = "us-west-2"
+	TERRAFORM_REMOTE_STATE_S3_BUCKET_NAME = "gruntwork-terragrunt-tests"
+	TEST_FIXTURE_PATH                     = "fixture/"
+)
+
+func TestTerragruntWorksWithLocalTerraformVersion(t *testing.T) {
+	if err := validateTerraformIsInstalled(t); err != nil {
+		t.Fatalf("A local instance of Terraform is not in the system PATH. Please install Terraform to continue.\n")
+	}
+
+	if err := validateS3BucketExists(t, TERRAFORM_REMOTE_STATE_S3_REGION, TERRAFORM_REMOTE_STATE_S3_BUCKET_NAME); err != nil {
+		t.Fatalf("The S3 Bucket in the .terragrunt file does not exist. %s", err)
+	}
+
+	if err := runTerragruntApply(); err != nil {
+		t.Fatalf("Failed to run terragrunt: %s", err)
+	}
+}
+
+// Run Terragrunt Apply directory in the test fixture path
+func runTerragruntApply() error {
+	os.Chdir(TEST_FIXTURE_PATH)
+
+	app := cli.CreateTerragruntCli("TEST")
+	err := app.Run(strings.Split("terragrunt apply", " "))
```

Comment:
Good call...I fixed those types of situations earlier and then had to revert them under the use of `assert` since the assert print to stdout gets swallowed when it's in a return, but in this case, that didn't apply. Fixed.